### ISSUE TITLE
Tradetrainer's _evaluate method does not support set_name "validation"

### ISSUE THREAD

**Issue by ieow** [ZhengyaoJiang/PGPortfolio#96]
_evaluate method in Tradetrainer is called with args {set_name : "validation"} in rollingtrainer line 32 is not supported by the method _evaluate.

_evaluate method only support set_name "test" and "training"

in tradetrainer.py line 74
``

    def _evaluate(self, set_name, *tensors):
        if set_name == "test":
            feed = self.test_set
        elif set_name == "training":
            feed = self.training_set
        else:
            raise ValueError()
        result = self._agent.evaluate_tensors(feed["X"],feed["y"],last_w=feed["last_w"],
                                              setw=feed["setw"], tensors=tensors)
        return result
``

in rollingtrainer.py line 32

``

    def __rolling_logging(self):
        fast_train = self.train_config["fast_train"]
        if not fast_train:
            tflearn.is_training(False, self._agent.session)

            v_pv, v_log_mean = self._evaluate("validation",
                                              self._agent.portfolio_value,
                                              self._agent.log_mean)
            t_pv, t_log_mean = self._evaluate("test", self._agent.portfolio_value, self._agent.log_mean)
            loss_value = self._evaluate("training", self._agent.loss)

            logging.info('training loss is %s\n' % loss_value)
            logging.info('the portfolio value on validation asset is %s\nlog_mean is %s\n' %
                         (v_pv,v_log_mean))
            logging.info('the portfolio value on test asset is %s\n mean is %s' % (t_pv,t_log_mean))
``
